src/components/PeoplePage.tsx

```diff
@@ -0,0 +1,138 @@
+import { Link, useParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { useEffect, useState } from 'react';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { Loader } from './Loader';
+
+export const PeoplePage: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [error, setError] = useState(false);
+
+  const { slug } = useParams();
+
+  useEffect(() => {
+    setLoading(true);
+
+    getPeople()
+      .then(setPeople)
+      .catch(() => {
+        setError(true);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+
+  const getMother = (person: Person) => {
+    return people.find(
+      parent => parent.name === person.motherName,
+    );
+  };
+
+  const getFather = (person: Person) => {
+    return people.find(
+      parent => parent.name === person.fatherName,
+    );
+  };
```

Comment:
Currently it is executed on every render, too slow. Better do it in `useEffect` after `getPeople` or in `useMemo`